Build error with AppleClang in contrib/libdivide

## Enhancement

```
[build] /Users/zanmato/dev/tiflash/contrib/libdivide/libdivide.h:2368:41: error: no member named 'denom' in 'divider<T, ALGO>'
[build]  2368 |         return div.denom.magic == other.denom.magic && div.denom.more == other.denom.more;
```

--- a/contrib/libdivide/libdivide.h
+++ b/contrib/libdivide/libdivide.h
@@ -2365,7 +2365,7 @@ class divider {
     T recover() const { return div.recover(); }
 
     bool operator==(const divider<T, ALGO> &other) const {
-        return div.denom.magic == other.denom.magic && div.denom.more == other.denom.more;
+        return div.denom.magic == other.div.denom.magic && div.denom.more == other.div.denom.more;
     }
 
     bool operator!=(const divider<T, ALGO> &other) const { return !(*this == other); }

